egs/wsj/s5/steps/data/reverberate_data_dir.py

```diff
@@ -0,0 +1,508 @@
+    rir_parser.add_argument('--receiver-position-id', type=str, default=None, help='receiver position id')
+    rir_parser.add_argument('--source-position-id', type=str, default=None, help='source position id')
+    rir_parser.add_argument('--rt60', type=float, default=None, help='RT60 is the time required for reflections of a direct sound to decay 60 dB.')
+    rir_parser.add_argument('--drr', type=float, default=None, help='Direct-to-reverberant-ratio of the impulse.')
+    rir_parser.add_argument('--probability', type=float, default=None, help='probability of the impulse.')
+    rir_parser.add_argument('rir_file_location', type=str, help='rir file location')
+
+    rir_list = []
+    rir_lines = map(lambda x: x.strip(), open(rir_list_file))
+    for line in rir_lines:
+        rir = rir_parser.parse_args(line.split())
+        setattr(rir, "iso_noise_list", [])
+        rir_list.append(rir)
+
+    return SmoothProbabilityDistribution(rir_list)
+
+
+# This function divides the global RIR list into local lists
+# according to the room where the RIRs are generated
+# It returns the room dictionary indexed by the room id
+# Its values are objects with two attributes: a local RIR list
+# and the probability of the corresponding room
+def MakeRoomDict(rir_list):
+    room_dict = {}
+    for rir in rir_list:
+        if rir.room_id not in room_dict:
+            # add new room
+            room_dict[rir.room_id] = lambda: None
+            setattr(room_dict[rir.room_id], "rir_list", [])
+            setattr(room_dict[rir.room_id], "probability", 0)
+        room_dict[rir.room_id].rir_list.append(rir)
+
+    # the probability of the room is the sum of probabilities of its RIR
+    for key in room_dict.keys():
+        room_dict[key].probability = sum(rir.probability for rir in room_dict[key].rir_list)
+
+    return room_dict
+
+
+# This function creates the point-source noise list 
+# and the isotropic noise list from the noise information file
+# Each noise item in the list contains the following attributes:
+# noise_id, noise_type, bg_fg_type, room_linkage, probability, noise_file_location
+# Please refer to the help messages in the parser for the meaning of these attributes
+def ParseNoiseList(noise_list_file):
+    noise_parser = argparse.ArgumentParser()
+    noise_parser.add_argument('--noise-id', type=str, required=True, help='noise id')
+    noise_parser.add_argument('--noise-type', type=str, required=True, help='the type of noise; i.e. isotropic or point-source', choices = ["isotropic", "point-source"])
+    noise_parser.add_argument('--bg-fg-type', type=str, default="background", help='background or foreground noise', choices = ["background", "foreground"])
+    noise_parser.add_argument('--room-linkage', type=str, default=None, help='required if isotropic, should not be specified if point-source.')
+    noise_parser.add_argument('--probability', type=float, default=None, help='probability of the noise.')
```

Comment:
Does the probability mass across point source noises and isotropic noises sum to 1 ?